src/users/user.service.ts

```diff
@@ -115,10 +114,16 @@ export class UserService {
 
   /**
    * Check user existence by email
+   * Check if user is deleted and active by email
    * @param emailDto
    */
   async checkEmail(emailDto: CheckEmailDto): Promise<boolean> {
     const foundUser = await this.userModel.find({ email: emailDto.email }).exec();
-    return foundUser.length > 0;
+    if (foundUser.length > 0) {
+      const user = await this.userModel.findOne({ email: emailDto.email }).exec();
+      const { isActivated, isDeleted } = user;
+      return !isActivated || isDeleted ? false : true;
```

Comment:
This logic is a little confusing. If you need to reverse the result to false : true, you can simply add a ! (exclamation) before all your logic and put the existing logic into a bracket.

Reply:
fixed it